### ISSUE TITLE
Request: Official tauri plugin for oauth

### ISSUE THREAD

**Issue by anurag-dhamala** [tauri-apps/plugins-workspace#265]
I'd like to request an official tauri plugin to support outh2 authentication where it would open the default browser of the user for authentication, and redirect back to the tauri-app once the authentication is complete. For that, seamless deep-linking should also be supported by tauri.

**Comment by Simon-Laux** [comment]
There is a deeplink plugin, what else would be needed for this use-case?

**Comment by Eveeifyeve** [comment]
> There is a deeplink plugin, what else would be needed for this use-case?

Does even deeplinking on android, ios and mac yet in runtime?

**Comment by FabianLars** [comment]
@Eveeifyeve apologies but could you try to re-phrase that sentence? I don't understand what you're saying/asking, sorry.

**Comment by Eveeifyeve** [comment]
> [@Eveeifyeve](https://github.com/Eveeifyeve) apologies but could you try to re-phrase that sentence? I don't understand what you're saying/asking, sorry.

According to the [tauri plugin website](https://tauri.app/plugin/deep-linking/) it does not support runtime deeplinking via both mobile platforms and macos. 

That is a problem for OAuth Plugins like atm, since I am making a plugin for minecraft authentification that uses deeplinking. 
But I can't since macos doesn't support runtime deeplinking according to your website.

**Comment by FabianLars** [comment]
Yeah but those are system limitations. We can't change that. On those platforms your app has to declare its deep link schemes upfront (in its "installer") and can't register new ones at runtime via the plugin's `register()` method.

I don't see how that would prevent the use in oauth flows though.

**Comment by Simon-Laux** [comment]
To my knowledge there are two kinds of "deep links":

## custom scheme

In the oauth2 flows that I have seen before you can use a custom scheme in the callback url. That would be the classical `myappname:<content>`, this is what FabianLars mentioned.


## "Universal Applink"

On some platforms there is also the possibility to register your app to handle web addresses to your website, though this process is generally more involved, like you need a special file on your website to authenticate that your app is allowed to handle it and you need to sign your app with a special entitlement (at least on macOS/iOS).

Platform wise (operating system support, not talking about what tauri supports here) this is available on android, iOS and macOS, I don't know about the other operating systems.

I made a pr to tao to add the handler code for handling such web addresses for macOS https://github.com/tauri-apps/tao/pull/1108

## Conclusion

I'd try the custom schemes first as the other method is way more complex and requires you to contribute to tauri to make it work among other things.

**Comment by FabianLars** [comment]
The official plugin supports universal applinks on ios and android.
On iOS it _only_ supports those kind of deep links and _not_ the custom schemes one.
On android the applinks are strongly recommended but custom schemes work as well since https://github.com/tauri-apps/plugins-workspace/pull/993.

**Comment by anurag-dhamala** [comment]
@Simon-Laux @FabianLars 

I was building a desktop app with Tauri back when the deeplink plugin didn't exist, so I created this issue requesting an OAuth plugin with functionality similar to deeplink. It looks like the deeplink plugin now supports what I originally needed, and it works just fine!

Thanks to the team.
